modules/core/04-channel/keeper/upgrade.go

```diff
@@ -106,20 +73,18 @@ func (k Keeper) ChanUpgradeTry(ctx sdk.Context, portID string, channelID string,
 		return 0, errorsmod.Wrapf(types.ErrInvalidChannelState, "expected one of [%s, %s], got %s", types.OPEN, types.INITUPGRADE, channel.State)
 	}
 
-	if proposedUpgradeChannel.Counterparty.PortId != channel.Counterparty.PortId ||
-		proposedUpgradeChannel.Counterparty.ChannelId != channel.Counterparty.ChannelId {
-		return 0, errorsmod.Wrap(types.ErrInvalidChannel, "counterparty port ID and channel ID cannot be upgraded")
-	}
+	// TODO: add check that if crossing hellos case, the currently set proposed upgrade == the passed in proposed upgrade fields (ie they did not change)
 
-	if !channel.Ordering.SubsetOf(proposedUpgradeChannel.Ordering) {
-		return 0, errorsmod.Wrap(types.ErrInvalidChannelOrdering, "channel ordering must be a subset of the new ordering")
+	// proposed upgrade fields must be equal on both sides of the upgrade
+	if reflect.DeepEqual(proposedUpgrade.ProposedUpgrade, counterpartyProposedUpgrade.ProposedUpgrade) {
+		return 0, errorsmod.Wrap(types.ErrChannelExists, "existing channel end is identical to proposed upgrade channel end")
 	}
 
-	if counterpartyUpgradeChannel.Ordering != proposedUpgradeChannel.Ordering {
-		return 0, errorsmod.Wrapf(types.ErrInvalidChannelOrdering, "channel ordering of counterparty channel and proposed channel must be equal")
+	if err := k.validateProposedUpgradeFields(ctx, channel, proposedUpgrade.ProposedUpgrade); err != nil {
+		return 0, err
 	}
 
-	connectionEnd, err := k.getUpgradeTryConnectionEnd(ctx, portID, channelID, channel)
+	connectionEnd, err := k.GetConnection(ctx, channel.ConnectionHops[0])
```

Comment:
if we move a check for the connection being open to `validateProposedUpgradeFields`, can we remove lines 87-97 (check for connectionEnd error and connectionEnd state being OPEN)